src/CommandHandler.cxx

```diff
@@ -2,8 +2,79 @@
 #include "helper.h"
 #include "utils.h"
 
+#include <future>
+
 namespace FileWriter {
 
+std::string find_filename(rapidjson::Document const &d) {
+  auto m1 = d.FindMember("file_attributes");
+  if (m1 != d.MemberEnd() && m1->value.IsObject()) {
+    auto m2 = m1->value.FindMember("file_name");
+    if (m2 != m1->value.MemberEnd() && m2->value.IsString()) {
+      return m2->value.GetString();
+    }
+  }
+  return std::string{"a-dummy-name.h5"};
+}
+
+std::string find_job_id(rapidjson::Document const &d) {
+  auto m = d.FindMember("job_id");
+  if (m != d.MemberEnd() && m->value.IsString()) {
+    return m->value.GetString();
+  }
+  return std::string{""};
+}
+
+std::string find_broker(rapidjson::Document const &d) {
+  auto m = d.FindMember("broker");
+  if (m != d.MemberEnd() && m->value.IsString()) {
+    auto s = std::string(m->value.GetString());
+    if (s.substr(0, 2) == "//") {
+      uri::URI u(s);
+      return u.host_port;
+    } else {
+      // legacy semantics
```

Comment:
If you mean get rid of the ``//legacy semantics`` part I'm not sure. The choice here depends on how we want to describe the broker in the command file. Will it be part of the ``topic`` field or will it be a field on its own? What if we keep it for the moment (maybe we can open an issue) and we wait until we discuss with the rest of the collaboration?

Reply:
It is not clear to me what your "// legacy semantics" comment means, is it because you have changed the json format?
If so, my point is that we really should not be afraid of losing support for a format which no one is using yet anyway.

Reply:
I see, if the decision is to support both then it is fine, but the comment is misleading.
"legacy" implied keeping something just for backwards compatibility, this is what concerned me.